substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -868,6 +868,11 @@ fn build_bloaty_blob(
 		// We don't want to call ourselves recursively
 		.env(crate::SKIP_BUILD_ENV, "");
 
+	let cargo_args = env::var(crate::WASM_BUILD_CARGO_ARGS).unwrap_or_default();
+	if !cargo_args.is_empty() {
+		build_cmd.args(cargo_args.split(" "));
```

Comment:
Doing a `split(" ")` here isn't necessarily correct as this won't handle shell escaping and will produce nonsensical result if someone tries to escape the arguments they pass here. Maybe that's not needed for the use case, but would probably be nice to have to prevent surprises in the future.

You could use the [shlex](https://docs.rs/shlex/1.3.0/shlex/index.html)::[split](https://docs.rs/shlex/1.3.0/shlex/fn.split.html#) crate to make this nicer, something like this should most likely work:

```suggestion
		let Some(args) = shlex::split(&cargo_args) else {
			build_helper::warning!("the {} environment variable is not a valid shell string", crate::WASM_BUILD_CARGO_ARGS);
			std::process::exit(1);
		};
		build_cmd.args(&args);
```

Reply:
I agree this can be an issue, but I thought initially that shell escaping should be a very distant use case. I am ok though with the idea of adding a shlex dependency to stop worrying about it.

Reply:
Done here: [a18681d](https://github.com/paritytech/polkadot-sdk/pull/5838/commits/a18681d79fee6aa2eb30620ffa670902b6ed23b9)